pkg/objstore/objstore.go

```diff
@@ -170,6 +170,23 @@ func DownloadDir(ctx context.Context, logger log.Logger, bkt BucketReader, src,
 	return nil
 }
 
+// Exists returns true, if file exists, otherwise false and nil error if presence IsObjNotFoundErr, otherwise false with
+// returning error.
+func Exists(ctx context.Context, bkt Bucket, src string) (bool, error) {
```

Comment:
I don't really like this helper. Having more than two ways of doing things is usually an antipattern. It's only used in single place, can we maybe make this quick helper , local function instead? Not a blocker

Reply:
why not?

https://github.com/improbable-eng/thanos/blob/7466e9091a600fa8029e3dea35fca5a8cc4e03c0/pkg/objstore/objstore.go#L47

you literally suggested to do the same

Reply:
Yea, but with explicit REMOVING statement ;p

Reply:
But you are right, let's leave it and  remove EXISTS from interface in another PRs, thanks for pointing this (: